kernel/dw1000.c

```diff
@@ -1335,17 +1331,20 @@ static int dw1000_ptp_settime(struct ptp_clock_info *ptp,
 			      const struct timespec64 *ts)
 {
 	struct dw1000 *dw = container_of(ptp, struct dw1000, ptp.info);
-	uint64_t ns;
+	struct hires_counter *tc = &dw->ptp.tc;
+	struct timehires time;
 
 	/* Convert to nanoseconds */
-	ns = timespec64_to_ns(ts);
+	time.tv_nsec = timespec64_to_ns(ts);
+	time.tv_frac = 0;
 
-	/* Reinitialise timecounter */
+	/* Reset timecounter */
 	mutex_lock(&dw->ptp.mutex);
-	timecounter_init(&dw->ptp.tc, &dw->ptp.cc, ns);
+	hires_counter_settime(tc, time);
 	mutex_unlock(&dw->ptp.mutex);
 
-	dev_dbg(dw->dev, "set time %llu ns\n", ns);
+	dev_dbg(dw->dev, "set time %llu ns\n", time.tv_nsec);
+
```

Comment:
Please remove stray blank line here (and elsewhere between dev_dbg() and return) for consistency with the rest of this file